Thirdweb/Thirdweb.Wallets/IThirdwebWallet.cs

```diff
@@ -16,11 +17,64 @@ public interface IThirdwebWallet
             where TDomain : IDomain;
         public Task<bool> IsConnected();
         public Task<string> SignTransaction(TransactionInput transaction, BigInteger chainId);
+        public Task<string> Authenticate(string domain, BigInteger chainId, string authPayloadPath = "/auth/payload", string authLoginPath = "/auth/login", HttpClient httpClient = null);
     }
 
     public enum ThirdwebAccountType
     {
         PrivateKeyAccount,
         SmartAccount
     }
+
+    [Serializable]
+    public struct LoginPayload
+    {
+        public LoginPayloadData payload;
+        public string signature;
+    }
+
+    [Serializable]
+    public class LoginPayloadData
+    {
+        [JsonProperty("type")]
+        public string Type { get; set; }
+
+        [JsonProperty("domain")]
+        public string Domain { get; set; }
+
+        [JsonProperty("address")]
+        public string Address { get; set; }
+
+        [JsonProperty("statement")]
+        public string Statement { get; set; }
+
+        [JsonProperty("uri", NullValueHandling = NullValueHandling.Ignore)]
+        public string Uri { get; set; }
+
+        [JsonProperty("version", NullValueHandling = NullValueHandling.Ignore)]
+        public string Version { get; set; }
+
+        [JsonProperty("chain_id", NullValueHandling = NullValueHandling.Ignore)]
+        public string ChainId { get; set; }
+
+        [JsonProperty("nonce", NullValueHandling = NullValueHandling.Ignore)]
+        public string Nonce { get; set; }
+
+        [JsonProperty("issued_at", NullValueHandling = NullValueHandling.Ignore)]
+        public string IssuedAt { get; set; }
+
+        [JsonProperty("expiration_time", NullValueHandling = NullValueHandling.Ignore)]
+        public string ExpirationTime { get; set; }
+
+        [JsonProperty("invalid_before", NullValueHandling = NullValueHandling.Ignore)]
+        public string InvalidBefore { get; set; }
+
+        [JsonProperty("resources", NullValueHandling = NullValueHandling.Ignore)]
+        public List<string> Resources { get; set; }
+
+        public LoginPayloadData()
+        {
+            Type = "evm";
```

Comment:
Don't need this anymore, we used to use it when we also supported Solana but it's not actually part of any spec

Reply:
I think we still use it in verify js code no?

Reply:
or at least check for it, I remember last week I glanced over some code that did the check 🤔

Reply:
well in this case we don't use it anyway so would only be a problem if someone hits old sdk with this payload type and it expects type, will remove

Reply:
nvm it stopped working on local server without it, keeping